client/utilities/hooks/useAsyncLoader.ts

```diff
@@ -39,7 +39,7 @@ export function useAsyncLoader<T>(
 			asyncFetch()
 				.then((response) => responseProcessor(response))
 				.then((data) => {
-					setData(data);
+					setData(data as T);
```

Comment:
this is basically the crux of the refactor - using `as` to assert it's the type, since we're not parsing it at all

Reply:
Would it make sense to add some validation here before casting to `T`? Is there a pattern for doing that in this codebase already? We use zod in some other projects.

Reply:
yes, absolutely. Seemed like a step too far for this PR - but I think zod was just added to the codebase recently